test/parallel-benchmark/mzcompose.py

```diff
@@ -1297,14 +1104,189 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
             gc.collect()
             gc.enable()
         finally:
-            failures.extend(
-                report(
-                    mz_string, scenario, state.measurements, start_time, args.guarantees
-                )
+            new_stats, new_failures = report(
+                mz_string,
+                scenario,
+                state.measurements,
+                start_time,
+                args.guarantees,
+                suffix,
             )
+            failures.extend(new_failures)
+            stats[scenario_name] = new_stats
+
+    if not args.mz_url:
+        c.kill("cockroach", "materialized", "testdrive")
+        c.rm("cockroach", "materialized", "testdrive")
+        c.rm_volumes("mzdata")
+
+    return stats, failures
+
+
+def check_regressions(
+    this_stats: dict[str, dict[str, Statistics]],
+    other_stats: dict[str, dict[str, Statistics]],
+) -> list[TestFailureDetails]:
+    failures: list[TestFailureDetails] = []
+
+    assert len(this_stats) == len(other_stats)
+
+    for scenario in this_stats.keys():
+        for query in this_stats[scenario].keys():
+            for stat in dir(this_stats[scenario][query]):
+                if stat not in REGRESSION_THRESHOLDS:
+                    continue
+
+                this_value = getattr(this_stats[scenario][query], stat)
+                other_value = getattr(other_stats[scenario][query], stat)
+                less_than = stat == "qps"
+                if (
+                    this_value < other_value / REGRESSION_THRESHOLDS[stat]
+                    if less_than
+                    else this_value > other_value * REGRESSION_THRESHOLDS[stat]
+                ):
+                    percent = (
+                        ((other_value - this_value) / other_value)
+                        if less_than
+                        else ((this_value - other_value) / (other_value))
+                    ) * 100
+                    # TODO: Table instead?
+                    failure = f"Scenario {scenario} regressed: {query}: {stat}: {this_value:.2f} {'<' if less_than else '>'} {other_value:.2f} ({percent:.2f}% regression)"
```

Comment:
Do we also want to print significant improvements?

Reply:
Yes, that's why I wanted to make it a table instead.